test/test-mc-range-edge-generation.c

```diff
@@ -261,6 +261,80 @@ _test_getEdgesDouble (_mongocrypt_tester_t *tester)
    }
 }
 
+
+#define MAX_DEC128_EDGES 128
+typedef struct {
+   mc_dec128 value;
+   mc_optional_dec128_t min;
+   mc_optional_dec128_t max;
+   int sparsity;
+   // expectEdges includes a trailing NULL pointer.
+   const char *expectEdges[MAX_DEC128_EDGES + 1];
+   const char *expectError;
+} Decimal128Test;
+#undef MAX_DEC128_EDGES
+
+static void
+_test_getEdgesDecimal128 (_mongocrypt_tester_t *tester)
+{
+   Decimal128Test tests[] = {
+#include "data/range-edge-generation/edges_decimal128.cstruct"
+   };
+
+   for (size_t i = 0; i < sizeof (tests) / sizeof (tests[0]); i++) {
+      const Decimal128Test *test = tests + i;
+      mongocrypt_status_t *const status = mongocrypt_status_new ();
+      mc_getEdgesDecimal128_args_t args = {
+         .value = test->value,
+         // Some edges specify min/max values, but we don't use them (yet)
+         //  .min = test->min,
+         //  .max = test->max,
+         .sparsity = (size_t) test->sparsity,
+      };
+      mc_edges_t *got = mc_getEdgesDecimal128 (args, status);
+
+      if (test->expectError != NULL) {
+         if (NULL != got) {
+            TEST_ERROR ("test %zu expected error, got success", i);
+         }
+         ASSERT_STATUS_CONTAINS (status, test->expectError);
+         mongocrypt_status_destroy (status);
+         continue;
+      }
+      ASSERT_OK_STATUS (got != NULL, status);
+
+      size_t numGot = mc_edges_len (got);
+      size_t numExpected = 0;
+      while (test->expectEdges[numExpected] != NULL) {
```

Comment:
I observe the following UBSAN runtime error during execution of `_test_getEdgesDecimal128`:
```
./test/test-mc-range-edge-generation.c:308:14: runtime error: index 129 out of bounds for type 'const char *const[129]'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ./test/test-mc-range-edge-generation.c:308:14 in
```
triggered by the first test in `./test/data/range-edge-generation/edges_decimal128.cstruct` which has `129` elements in `.expectEdges`, exceeding `MAX_DEC128_EDGES` and overwriting the expected `NULL` terminator in `test->expectEdges`.

Reply:
I can repro this one. Not sure how I missed it.